### PR TITLE
[iOS] carousel view  element for iOS 

### PR DESCRIPTION
**Pull request by guptaabhishe**
# Description

This PR aims  for carousel view for iOS. 

This is video for carousel view 

https://github.com/user-attachments/assets/43f8ee5b-ea37-4027-8b34-de808801a1ab

### CODE REVIEW

**Review by wyAbhishek — CHANGES_REQUESTED** [review]
- We should break big renderer into smaller reusable chunks
- We need to change LayoutHelper for Carousel and CarouselPage element.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselPageContainerView.mm [hunk 1]
```diff
@@ -0,0 +1,63 @@
+//
+//  ACRCarouselPageContainerView.mm
+//  AdaptiveCards
+//
+//  Created by Abhishek Gupta on 25/09/24.
+//  Copyright © 2024 Microsoft. All rights reserved.
+//
+
+#import "ACOBaseCardElementPrivate.h"
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselPageContainerView.mm:9]: Do we need all these Headers? Looks like many things can be removed.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    
+    self.carousel = carousel;
+    
+    
+    ACRColumnView *carouselViewContainer = [[ACRColumnView alloc] initWithStyle:(ACRContainerStyle)carousel->GetStyle()
+                                                        parentStyle:[viewGroup style]
+                                                         hostConfig:acoConfig
+                                                          superview:viewGroup];
+    
+    for(auto carouselPage: carousel->GetItems()) {
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm:76]: We are following objective-C styled code formatting. 
for(..)
{
} 

if()
{
}
else
{
}

Please check in other places as well

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    
+    self.carouselPageViewList[0].hidden = NO;
+    self.carouselPageViewIndex = 0;
+    
+    std::string selctedPageControlTintColor = config->GetPageControlConfig().selectedTintColor;
+    std::string unselctedPageControlTintColor = config->GetPageControlConfig().unselectedTintColor;
+    
+    ACRPageControlConfig *pageControlConfig = [[ACRPageControlConfig alloc] initWithNumberOfPages:self.carouselPageViewList.count
+                                                                                     displayPages:@7
+                                                                                 selctedTintColor:[ACOHostConfig convertHexColorCodeToUIColor:selctedPageControlTintColor]
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm:104]: nit: indentation looks off

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+            break;
+        case PageAnimation::CrossFade:
+            [self crossfadeFromView:oldView toView:newView];
+        case PageAnimation::None:
+            oldView.hidden = YES;
+            newView.hidden = NO;
+        default:
+            break;
+    }
+    self.carouselPageViewIndex = newCarouselPageViewIndex;
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm:176]: I think, This code(and similar below) should be part of carouselPagesContainerView?

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselPageContainerView.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+#import "ACOHostConfig.h"
+#import "ACRContentStackView.h"
+#import "ACRView.h"
+#import <UIKit/UIKit.h>
+
+@interface ACRCarouselPageContainerView : UIView
+
+-(instancetype) initWithCarouselPageViewList:(NSMutableArray<UIView *> *)initWithCarouselPageViewList;
+
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselPageContainerView.h:17]: Not able to find method like setCurrentPage here? Similar to PageControl

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselPageView.mm [hunk 6]
```diff
@@ -0,0 +1,183 @@
+//
+//  ACRCarouselPageView.mm
+//  AdaptiveCards
+//
+//  Created by Abhishek Gupta on 25/09/24.
+//  Copyright © 2024 Microsoft. All rights reserved.
+//
+
+#import "ACOBaseCardElementPrivate.h"
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselPageView.mm:9]: Too many headers, please check everywhere.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRPageControl.mm [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    NSInteger currentPage = MIN(MAX(self.currentPage, 0), self.numberOfPages - 1);
+    
+    CGPoint point = CGPointMake( floor((self.bounds.size.width - self.requiredLength) / 2.0 ), 0.0);
+    
+    point.x -= _spacing *(visibleViewStartIndex);
+    
+    for(NSInteger i =0 ; i< self.numberOfPages; i++) {
+        
+        BOOL isSelectedPage = (i == currentPage);
+
```
Comment (wyAbhishek) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRPageControl.mm:183]: nit: extra space

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm (hunk 3 above)
Comment (guptaabhishe) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRCarouselView.mm:104]: It should be correct all ':' are in same line vertically

**Review by wyAbhishek — APPROVED** [review]
Please add more details to description.

**Review by prpatwa — APPROVED** [review]

**Pull request merged**